### PR TITLE
fix(build): special case inconsistently-cased paths in LLVM build

### LINKED ISSUE #4535: LLVM doesn't build on MacOS
```
$ bazel build @org_llvm//:clangBasic
INFO: Analyzed target @org_llvm//:clangBasic (4 packages loaded, 2150 targets configured).
INFO: Found 1 target...
ERROR: /private/var/tmp/_bazel_jedmonds/e03751731153a67e0a8638c48dafe42b/external/org_llvm/BUILD.bazel:208:1: C++ compilation of rule '@org_llvm//:LLVMTextAPI' failed: I/O exception during sandboxed execution: /private/var/tmp/_bazel_jedmonds/e03751731153a67e0a8638c48dafe42b/sandbox/darwin-sandbox/5825/execroot/io_kythe/external/org_llvm/include/llvm/TextAPI/Elf/ELFStub.h (File exists)
Target @org_llvm//:clangBasic failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 207.156s, Critical Path: 34.17s
INFO: 380 processes: 380 darwin-sandbox.
FAILED: Build did NOT complete successfully
```

Reverting commit 3c3498abb616f8140c5fbfa5f06b023da6b69a07 fixes the issue. (Note I'm building @org_llvm//:clangBasic because it's faster than building //kythe/release but both exhibit the same issue and both successfully compile with 3c3498abb616f8140c5fbfa5f06b023da6b69a07 reverted)

### PR DESCRIPTION
**Pull request by shahms**
Fixes #4535

### CODE REVIEW

File: tools/build_rules/llvm/llvm.bzl [hunk 1]
```diff
@@ -123,6 +129,15 @@ def _llvm_library(ctx, name, srcs, hdrs = [], deps = [], additional_header_dirs
         for s in srcs
         if paths.dirname(s)
     }.keys()
+
+    # Upstream LLVM has a single inconsistent-case header directory
+    # which causes breakages on macOS.
+    # Adjust the case here if we encounter it.
+    # https://github.com/kythe/kythe/issues/4535
+    additional_header_dirs = [
+        _replace_suffix(dir, {"/Elf": "/ELF", "/AsmParser": "/ASMParser"})
```
Comment (johnedmonds) [inline tools/build_rules/llvm/llvm.bzl:138]: I think /AsmParser and /ASMParser needs to be reversed so the line would be:

```
_replace_suffix(dir, {"/Elf": "/ELF", "/ASMParser": "/AsmParser"})
```

File: tools/build_rules/llvm/llvm.bzl [hunk 2]
```diff
@@ -123,6 +129,15 @@ def _llvm_library(ctx, name, srcs, hdrs = [], deps = [], additional_header_dirs
         for s in srcs
         if paths.dirname(s)
     }.keys()
+
+    # Upstream LLVM has a single inconsistent-case header directory
```
Comment (johnedmonds) [inline tools/build_rules/llvm/llvm.bzl:133]:
```suggestion
    # Upstream LLVM has some inconsistent-case header directories
```

**Comment by shahms** [comment]
Ugh, I knew I shouldn't have done that.

@johnedmonds any chance you can go to https://cla.developers.google.com/ and sign the Google CLA? Our compliance bot is picky and because I applied the changes from the web UI rather than manually it's going to attribute them to you.  If you don't want to sign the CLA I can (probably?) revert the auto-commits and make the fixes myself.

**Comment by johnedmonds** [comment]
Sorry about that. Any chance you could change the author to jedmonds@spotify.com which has signed the CLA? (I just changed the default address on my profile page if that helps.)

**Comment by shahms** [comment]
I don't believe I can (or at least, have no idea how I would), but I believe you should be able to add the @gmail.com address at the linked page?

**Comment by johnedmonds** [comment]
Done. Signed the CLA.

**Pull request merged**